plugin.cpp

```diff
@@ -241,6 +259,8 @@ void plugin_ingest(PLUGIN_HANDLE *handle,
 
 	// Just get all the readings in the readingset
 	const vector<Reading *>& readings = origReadingSet->getAllReadings();
+
+	static const std::set<std::string> validDpTypes{"FLOAT", "INTEGER", "STRING", "FLOAT_ARRAY", "DP_DICT", "DP_LIST", "IMAGE", "DATABUFFER", "2D_FLOAT_ARRAY", "NUMBER", "NON-NUMERIC"};
```

Comment:
These don't match what was in the requirements. We should be using user friendly names and not those that are in the internal enumeration in the code.

DATABUFFER houd be buffer.

It's good that you have added more types, but DP_LIST and DP_DICT for example have little meaning to the average user.  Maybe user array to remove all array data, nested to remove DP_DICT types. FLOAT_ARRAY can be simply array, which would probably also remove the 2D arrays and 2D_FLOAT_ARRAY could be 2D Array.

Reply:
I am converting buffer to DATABUFFER and floating to FLOAT  before going for validity check of type.

Reply:
an array will remove float array or 2D_float array ?

Reply:
implementing like array for float array , 2DArray for  2D_FLOAT_ARRAY  and user _array for removing both float and 2DFloatArrays

Reply:
fixed.